### PR TITLE
Fix Ivar.fill bug in Transition_frontier_controller

### PR DESCRIPTION
**Pull request by georgeee**
When bootstrap is called twice over the Mina node's execution, the node would exit fatally with `Ivar.fill bug is here!`. This is absolutely pointless behavior and I believe it may be causing additional restarts when Mina node's catchup fails to start quickly.

Fix: instead of iterating the `clear_reader` which doesn't make sense, just read once.

Explain how you tested your changes:
* Ran a local two-node network with 25-second slot and 48 slots per epoch (`k=2`)
   * Two transactions per minute are generated int he network
   * In such a network non-BP node frequently falls out of sync because BP is not always able to finish creation of a block within 25 seconds (on my laptop), hence bootstraps happening after catchup are frequent and bug can be spotted in the first 30 minutes of network's operation

Checklist:

- [x] Dependency versions are unchanged
  - Notify Velocity team if dependencies must change in CI
- [ ] Modified the current draft of release notes with details on what is completed or incomplete within this project
- [x] Document code purpose, how to use it
  - Mention expected invariants, implicit constraints
- [x] Tests were added for the new behavior
  - Document test purpose, significance of failures
  - Test names should reflect their purpose
- [x] All tests pass (CI will check this if you didn't)
- [x] Serialized types are in stable-versioned modules
- [x] Does this close issues? None

### CODE REVIEW

**Comment by georgeee** [comment]
!ci-build-me

**Review by ghost-not-in-the-shell — APPROVED** [review]

**Review by rbonichon — APPROVED** [review]
LGTM

File: src/lib/transition_frontier_controller/transition_frontier_controller.ml [hunk 1]
```diff
@@ -116,15 +116,17 @@ let run ~logger ~trust_system ~verifier ~network ~time_controller
...
-      if Ivar.is_full clean_up_catchup_scheduler then
-        [%log error] "Ivar.fill bug is here!" ;
-      Ivar.fill clean_up_catchup_scheduler () )
-  |> don't_wait_for ;
+  upon (Strict_pipe.Reader.read clear_reader) (function
+    | `Eof ->
+        [%log error] "unexpected eof on clear_reader"
+    | `Ok _ ->
+        let open Strict_pipe.Writer in
+        kill valid_transition_writer ;
```
Comment (mrmr1993) [inline src/lib/transition_frontier_controller/transition_frontier_controller.ml:124]: Shouldn't we `kill` the pipes regardless? It should be safe either way.

Comment (georgeee) [reply src/lib/transition_frontier_controller/transition_frontier_controller.ml:124]: I don't know how the first case could ever trigger, honestly, so I'm fine to proceed with your suggestion @mrmr1993

**Comment by georgeee** [comment]
!ci-build-me

**Review by mrmr1993 — APPROVED** [review]

**Comment by mrmr1993** [comment]
!ci-build-me

**Pull request merged**